File: kirun-java/src/test/java/com/fincity/nocode/kirun/engine/function/system/array/FillTest.java
package com.fincity.nocode.kirun.engine.function.system.array;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertThrows;

import java.util.Map;

import org.junit.jupiter.api.Test;

import com.fincity.nocode.kirun.engine.runtime.FunctionExecutionParameters;
import com.google.gson.JsonArray;
import com.google.gson.JsonPrimitive;

class FillTest {

	@Test
	void testInternalExecute() {

		Fill fill = new Fill();

		FunctionExecutionParameters fep = new FunctionExecutionParameters();
		JsonArray array = new JsonArray();
		array.add(0);
		array.add(1);

		fep.setArguments(Map.of("source", array, "element", new JsonPrimitive(3)))
		        .setContext(Map.of())
		        .setOutput(Map.of());

		fill.execute(fep);

		JsonArray finArray = new JsonArray();
		finArray.add(new JsonPrimitive(3));
		finArray.add(new JsonPrimitive(3));

		assertEquals(finArray, array);

		fep.setArguments(Map.of("source", array, "element", new JsonPrimitive(5), "srcFrom", new JsonPrimitive(2),
		        "length", new JsonPrimitive(5)))
		        .setContext(Map.of())
		        .setOutput(Map.of());

		finArray = new JsonArray();
		finArray.add(new JsonPrimitive(3));
		finArray.add(new JsonPrimitive(3));
		finArray.add(new JsonPrimitive(5));
		finArray.add(new JsonPrimitive(5));
		finArray.add(new JsonPrimitive(5));
		finArray.add(new JsonPrimitive(5));
		finArray.add(new JsonPrimitive(5));
		
		fill.execute(fep);

		assertEquals(finArray, array);
		
		fep.setArguments(Map.of("source", array, "element", new JsonPrimitive(25), "srcFrom", new JsonPrimitive(5)))
		        .setContext(Map.of())
		        .setOutput(Map.of());

		finArray = new JsonArray();
		finArray.add(new JsonPrimitive(3));
		finArray.add(new JsonPrimitive(3));
		finArray.add(new JsonPrimitive(5));
		finArray.add(new JsonPrimitive(5));
		finArray.add(new JsonPrimitive(5));
		finArray.add(new JsonPrimitive(25));
		finArray.add(new JsonPrimitive(25));
		
		fill.execute(fep);

		assertEquals(finArray, array);
		
		fep.setArguments(Map.of("source", array, "element", new JsonPrimitive(20), "srcFrom", new JsonPrimitive(-1)))
        .setContext(Map.of())
        .setOutput(Map.of());
		
		
		assertThrows(IndexOutOfBoundsException.class, () -> fill.execute(fep));
	}

}

File: kirun-js/__tests__/engine/function/system/array/FillTest.ts
import { Fill } from '../../../../../src/engine/function/system/array/Fill';
import { FunctionExecutionParameters } from '../../../../../src/engine/runtime/FunctionExecutionParameters';
import { MapUtil } from '../../../../../src/engine/util/MapUtil';
import { KIRunFunctionRepository, KIRunSchemaRepository } from '../../../../../src';

test('Fill Test', async () => {
    let fill: Fill = new Fill();

    let fep: FunctionExecutionParameters = new FunctionExecutionParameters(
        new KIRunFunctionRepository(),
        new KIRunSchemaRepository(),
    );
    let array = [0, 1];

    fep.setArguments(MapUtil.of('source', array as any, 'element', 3));

    array = (await fill.execute(fep)).allResults()[0].getResult().get('result');
    expect(array).toStrictEqual([3, 3]);

    fep.setArguments(MapUtil.of('source', array as any, 'element', 5, 'srcFrom', 2, 'length', 5))
        .setContext(MapUtil.of())
        .setSteps(MapUtil.of());

    let finArray = [3, 3, 5, 5, 5, 5, 5];

    array = (await fill.execute(fep)).allResults()[0].getResult().get('result');
    expect(array).toMatchObject(finArray);

    fep.setArguments(MapUtil.of('source', array as any, 'element', 25, 'srcFrom', 5));

    finArray = [3, 3, 5, 5, 5, 25, 25];

    expect((await fill.execute(fep)).allResults()[0].getResult().get('result')).toMatchObject(
        finArray,
    );

    fep.setArguments(MapUtil.of('source', array as any, 'element', 20, 'srcFrom', -1));

    await expect(fill.execute(fep)).rejects.toThrow();
});
